packages/gestalt-codemods/generic-codemods/renameComponent.js

```diff
@@ -5,76 +5,77 @@
  *
  * OPTIONS:
- * --previousName: current component name to be replaced
- * --nextName: new component name to replace with
+ * --componentName: current component name to be replaced
+ * --nextComponentName: new component name to replace with
  */
 
 import {
-  getImports,
-  getJSX,
+  getGestaltImport,
+  getComponentIdentifierByName,
   getLocalImportedName,
+  filterJSXByTargetLocalName,
   initialize,
-  isGestaltImport,
-  matchesComponentName,
-  replaceImportedName,
-  replaceImportNodePath,
-  renameJSXElement,
   saveToSource,
-  sortImportedNames,
-} from './deprecatedUtils.js';
+  buildReplaceWithRenamedComponent,
+  buildReplaceWithRenamedImport,
+} from './utils.js';
 import { type FileType, type ApiType } from './flowtypes.js';
+// import { describe } from 'jscodeshift-helper';
 
-type OptionsType = {| previousName: string, nextName: string |};
+type OptionsType = {|
+  componentName: string,
+  nextComponentName: string,
+|};
 
-function transform(fileInfo: FileType, api: ApiType, options: OptionsType): ?string {
-  const { previousName, nextName } = options;
+function transform(fileInfo: FileType, api: ApiType, options: OptionsType): ?string | null {
+  const { componentName, nextComponentName } = options;
 
   const { j, src } = initialize({ api, fileInfo });
 
-  let targetLocalImportedName;
+  const gestaltImportCollection = getGestaltImport({ src, j });
 
-  getImports({ src, j }).forEach((nodePath) => {
-    const { node: importDeclaration } = nodePath;
+  if (gestaltImportCollection.size() === 0) return null;
 
-    if (!isGestaltImport({ importDeclaration })) return;
+  const componentIdentifierCollection = getComponentIdentifierByName({
+    j,
+    gestaltImportCollection,
+    componentName,
+  });
 
-    targetLocalImportedName = getLocalImportedName({
-      importDeclaration,
-      importedName: previousName,
-    });
+  if (componentIdentifierCollection.size() === 0) return null;
 
-    if (!targetLocalImportedName) return;
+  const targetLocalName = getLocalImportedName({
+    importSpecifierCollection: componentIdentifierCollection,
+  });
 
-    const newImportSpecifiers = replaceImportedName({
-      j,
-      importDeclaration,
-      previousComponentName: previousName,
-      nextComponentName: nextName,
-    });
+  const matchedJSXCollection = filterJSXByTargetLocalName({
+    src,
+    j,
+    targetLocalName,
+  });
 
-    const newSortedImportSpecifiers = sortImportedNames({ importSpecifiers: newImportSpecifiers });
+  if (matchedJSXCollection.size() === 0) return null;
 
-    replaceImportNodePath({
-      j,
-      nodePath,
-      importSpecifiers: newSortedImportSpecifiers,
-      importPath: 'gestalt',
-    });
+  const replaceWithModifiedCloneCallback = buildReplaceWithRenamedComponent({
+    nextComponentName,
   });
 
-  getJSX({ src, j }).forEach((nodePath) => {
-    const { node: JSXNode } = nodePath;
+  for (let idx = matchedJSXCollection.size() - 1; idx >= 0; idx -= 1) {
```

Comment:
No, because I have to go from children to parent. This is why. the replaceWith function replaces all nodes, but it includes parents and children. If the parent contains the children, it gets overriden and the children doesnt include the changes, reverting the order, children first, then parent, I was able to persist the changes in the children.

Reply:
Cool, makes sense. Can you document that in the code here?